src/render/src/lib.rs

```diff
@@ -143,3 +148,80 @@ pub mod pso;
 pub mod shade;
 /// Convenience macros
 pub mod macros;
+*/
+
+use std::ops;
+use std::sync::mpsc;
+
+pub struct Gpu<B: Backend> {
+    device: Device<B>,
+    queue: CommandQueue<B, core::General>,
+    garbage: handle::GarbageReceiver<B>,
+}
+
+impl<B: Backend> Gpu<B> {
+    pub fn new<A: core::Adapter<B>>(adapter: &A) -> Self {
+        let queue_descs = adapter.get_queue_families()
+            .iter()
+            .map(|&(ref family, qtype)| (family, qtype, family.num_queues()) )
+            .collect::<Vec<_>>();
+        
+        let core::Gpu { device, mut general_queues, heap_types, memory_heaps, .. } =
+            adapter.open(&queue_descs);
+        
+        let queue = general_queues.remove(0);
+
+        let (garbage_sender, garbage_receiver) = handle::garbage_channel();
+
+        Gpu {
+            device: Device {
+                device,
+                heap_types,
+                memory_heaps,
+                garbage: garbage_sender,
+            },
+            queue,
+            garbage: garbage_receiver,
+        }
+    }
+
+    pub fn cleanup(&mut self) {
+        use core::Device;
+
+        let dev = &mut self.device.device;
+        for garbage in self.garbage.try_iter() {
+            use handle::Garbage::*;
+            match garbage {
+                ShaderLib(sl) => dev.destroy_shader_lib(sl),
+                Buffer(b) => dev.destroy_buffer(b),
+                Image(i) => dev.destroy_image(i),
+                RenderTargetView(rtv) => dev.destroy_render_target_view(rtv),
+                DepthStencilView(dsv) => dev.destroy_depth_stencil_view(dsv),
+                ConstantBufferView(cbv) => dev.destroy_constant_buffer_view(cbv),
+                ShaderResourceView(srv) => dev.destroy_shader_resource_view(srv),
+                UnorderedAccessView(uav) => dev.destroy_unordered_access_view(uav),
+                Sampler(s) => dev.destroy_sampler(s),
+            }
+        }
+    }
+}
+
+impl<B: Backend> ops::Deref for Gpu<B> {
```

Comment:
We might rather expose explicit `get_device`/`ref_device`/..(?) functions than `Deref`.

Reply:
This will become a bit more verbose for users that are not actually using the `Device` split, but I agree that the use of `Deref` might be a bit abusive here.